vm/src/obj/objtuple.rs

```diff
@@ -257,10 +257,10 @@ impl PyValue for PyTupleIterator {
 impl PyTupleIterator {
     #[pymethod(name = "__next__")]
     fn next(&self, vm: &VirtualMachine) -> PyResult {
-        if self.position.get() < self.tuple.as_slice().len() {
-            let ret = self.tuple.as_slice()[self.position.get()].clone();
-            self.position.set(self.position.get() + 1);
-            Ok(ret)
+        let pos = self.position.load();
```

Comment:
Again lets use `fetch_add` here. In this solution we can have 2 thread getting the same value.